### PR TITLE
Correctly handle Javadoc method link

### LINKED ISSUE #159: Javadoc directive for a method links to the class page, not the method section of the page
http://developer.lightbend.com/docs/paradox/latest/features/linking.html#javadoc-directive

The doc says :

`@javadoc[Http](akka.http.javadsl.Http#shutdownAllConnectionPools--)`  

will resolve to 
http://doc.akka.io/japi/akka-http/10.0.0/akka/http/javadsl/Http.html#shutdownAllConnectionPools--.
_(it jumps to the **method section** of the class page)_

however, it actually doesn't.

I tried it in [my test project](https://github.com/richardimaoka/paradoxtest/blob/841f81dd033b912f25fa07072f66b1883782294b/src/main/paradox/index.md#L1),

2017-10-15_06h27_07

 and the actual link I got was (note `?` in the middle):

http://doc.akka.io/japi/akka-http/10.0.0/?akka/http/javadsl/Http.html#shutdownAllConnectionPools--.
_(it jumps to **the class page**, **not** the method section)_

### PR DESCRIPTION
**Pull request by richardimaoka**
Close #159

### CODE REVIEW

File: core/src/test/scala/com/lightbend/paradox/markdown/JavadocDirectiveSpec.scala [hunk 1]
```diff
@@ -46,7 +46,7 @@ class JavadocDirectiveSpec extends MarkdownBaseSpec {
 
   it should "handle method links correctly" in {
     markdown("@javadoc[File.pathSeparator](java.io.File#pathSeparator)") shouldEqual
-      html("""<p><a href="https://docs.oracle.com/javase/8/docs/api/?java/io/File.html#pathSeparator">File.pathSeparator</a></p>""")
+      html("""<p><a href="https://docs.oracle.com/javase/8/docs/api/java/io/File.html#pathSeparator">File.pathSeparator</a></p>""")
```
Comment (richardimaoka) [inline core/src/test/scala/com/lightbend/paradox/markdown/JavadocDirectiveSpec.scala]: (current) https://docs.oracle.com/javase/8/docs/api/?java/io/File.html#pathSeparator 
jumps to the class page,

(expected) https://docs.oracle.com/javase/8/docs/api/java/io/File.html#pathSeparator
correctly jumps to the method section of the page

Comment (jonas) [reply core/src/test/scala/com/lightbend/paradox/markdown/JavadocDirectiveSpec.scala]: The original behaviour of linking to the javadoc with frames comes from this suggestion by @jroper : https://github.com/lightbend/paradox/pull/52#issuecomment-258751524

I am not sure if there even exists a way to link to methods when opening with frames.

Comment (jroper) [reply core/src/test/scala/com/lightbend/paradox/markdown/JavadocDirectiveSpec.scala]: As @jonas points out, this change results in a terrible user experience. Out of the box there doesn't exist a way to link to methods when opening with frames, but we have this Javascript code that gets added to every page in the Lagom Javadocs:

https://github.com/lagom/lagom/blob/fde496bc089157e40a3a132a3ae1cf0d5f1111d4/project/Doc.scala#L63-L79

This then gets configured here:

https://github.com/lagom/lagom/blob/fde496bc089157e40a3a132a3ae1cf0d5f1111d4/project/Doc.scala#L134

That way, you can link to methods in the docs, and still have it open with frames.

Comment (richardimaoka) [reply core/src/test/scala/com/lightbend/paradox/markdown/JavadocDirectiveSpec.scala]: Thanks, let me look at how lagom works for this.

**Comment by ennru** [comment]
No activity on this for a more than a year, closing.

**Pull request closed without merging**